src/app/dataset-view/additional-components/metadata-component/components/form-components/schema-field/schema-field.component.spec.ts

```diff
@@ -0,0 +1,160 @@
+                    ]),
+                }),
+            ]),
+        });
+        component.controlName = "schema";
+        fixture.detectChanges();
+    });
+
+    it("should create", () => {
+        expect(component).toBeTruthy();
+    });
+
+    it("should check add row to form array", () => {
+        const item: SchemaType = { name: "id", type: "BIGINT" };
+        component.focus$.next(item);
+        expect(component.items.length).toBe(1);
+        emitClickOnElement(fixture, '[data-test-id="add-row-button"]');
+        expect(component.items.length).toBe(2);
+    });
+
+    it("should check delete row to form array", () => {
+        expect(component.items.length).toBe(1);
+        emitClickOnElement(fixture, '[data-test-id="delete-row-button-0"]');
+        expect(component.items.length).toBe(0);
+    });
+
+    it("should check unique validation message", () => {
+        component.addRow();
+        fixture.detectChanges();
+        const controlName_1 = findElementByDataTestId(
+            fixture,
+            "name-control-1",
+        ) as HTMLInputElement;
+        controlName_1.value = "id";
+        controlName_1.dispatchEvent(new Event("input"));
+        fixture.detectChanges();
+        expect(component.nameControlError(1)).toBe("Name is not unique");
+    });
+
+    it("should check required validation message", () => {
+        component.addRow();
+        fixture.detectChanges();
+        const controlName_1 = findElementByDataTestId(
+            fixture,
+            "name-control-1",
+        ) as HTMLInputElement;
+        controlName_1.value = "";
+        controlName_1.dispatchEvent(new Event("input"));
+        fixture.detectChanges();
+        expect(component.nameControlError(1)).toBe("Name is required");
+    });
+
+    it("should check noneOf validation message", () => {
+        component.addRow();
+        fixture.detectChanges();
+        const controlName_1 = findElementByDataTestId(
+            fixture,
+            "name-control-1",
+        ) as HTMLInputElement;
+        controlName_1.value = "STRING";
+        controlName_1.dispatchEvent(new Event("input"));
+        fixture.detectChanges();
+        expect(component.nameControlError(1)).toBe("Type cannot be name");
+    });
+
+    it("should check swap row", () => {
+        component.addRow();
+        fixture.detectChanges();
+        const controlName_1 = findElementByDataTestId(
+            fixture,
+            "name-control-1",
+        ) as HTMLInputElement;
+        controlName_1.value = "name";
+        controlName_1.dispatchEvent(new Event("input"));
+        const typeName_1 = findElementByDataTestId(
+            fixture,
+            "type-control-1",
+        ) as HTMLInputElement;
+        typeName_1.value = "STRING";
+        typeName_1.dispatchEvent(new Event("input"));
+        fixture.detectChanges();
+        expect(component.items.controls[0].get("name")?.value).toBe("id");
+
```

Comment:
Difficult to read. Maybe it would look better with the helpers I've mentioned above.

Reply:
Yes, refactored with new helpers.